### ISSUE TITLE
Machine testing of freshness (serviceLevels)

### ISSUE THREAD

**Issue by john-bunge-ed** [datacontract/datacontract-cli#261]
Question:

Is the _freshness_ parameter in the _serviceLevels_ object supposed to be machine testable?

I applied `test()` to my datacontract object (see below), however no automated test results on _freshness_ could be retrieved from the _.checks_ or _.logs_ elements.

```
datacontract_obj = DataContract(
    data_contract_file='datacontract.yaml',
    spark=spark)
datacontract_run = datacontract_obj.test()

datacontract_run.checks

datacontract_run.logs
```

_freshness_ was specified in the _datacontract.yaml_ as follows. Column _timestamp_col_ is of timestamp format.

```
servicelevels:
  freshness:
    description: The maximum age of the youngest entry is 1 hour
    threshold: 1h
    timestampField: table_name.timestamp_col
```

**Comment by jochenchrist** [comment]
Hi @john-bunge-ed,
this is currently not yet implemented, but would totally make sense.
Current workaround would be to add the freshness-test as additional quality test.

**Comment by john-bunge-ed** [comment]
@jochenchrist
Where in the codebase would you suggest to add such an additional quality test?

**Comment by jochenchrist** [comment]
next to this:
https://github.com/datacontract/datacontract-cli/blob/main/datacontract/export/sodacl_converter.py#L15

**Comment by john-bunge-ed** [comment]
thanks @jochenchrist

**Comment by jochenchrist** [comment]
closed as duplicate to #407
